### PR TITLE
Move Trusty e2e testing jobs to job-configs

### PR DESCRIPTION
**Pull request by wonderfly**
@spxtr jenkins-jobs test job-configs/ spits out the XML and it looks sane. How do I test it works on Jenkins before merging it in?

cc @ihmccreery @ixdy @andyzheng0831

### CODE REVIEW

File: hack/jenkins/job-configs/kubernetes-e2e-trusty.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        - mail-watcher
+    publishers:
+        - claim-build
+        - junit-publisher
+        - log-parser
+        - email-ext:
+            recipients: "{emails}"
+        - description-setter:
+            regexp: KUBE_GCE_MINION_IMAGE=(.*)
+        - groovy-postbuild: |
```
Comment (spxtr) [inline hack/jenkins/job-configs/kubernetes-e2e-trusty.yaml]: This is giving me a warning. I think you're supposed to use the `script` parameter as documented [here](http://docs.openstack.org/infra/jenkins-job-builder/publishers.html#publishers.groovy-postbuild).

`WARNING:jenkins_jobs.modules.publishers:groovy-postbuild:You use depricated configuration, please follow documentation to change configuration. It is not going to be supported in future releases!`

File: hack/jenkins/job-configs/kubernetes-e2e-trusty.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            timeout: '{timeout}'
+            fail: true
+        - timestamps
+        - workspace-cleanup
+
+- project:
+    name: kubernetes-e2e-gce-trusty
+    test-owner: 'wonderfly@google.com'
+    branch: 'release-1.1'
+    emails:
```
Comment (spxtr) [inline hack/jenkins/job-configs/kubernetes-e2e-trusty.yaml]: This should be a single comma-separated list. Ugly, I know, but it just puts it straight into the Jenkins config which expects it.

`emails: adityakali@google.com,ameyd@google.com,...`

**Comment by spxtr** [comment]
Two small changes, looks good otherwise.

Thanks for doing this :)

**Comment by spxtr** [comment]
Before merging, backup the XML configs from Jenkins for the current jobs. Manually merge, then make sure things look right. If they don't, POST the XML to Jenkins' remote API and it'll reset them. You only need to do this the first time, since once they're in source control we can just roll back breaking changes.

I'd be happy to do this for you for these jobs.

File: hack/jenkins/job-configs/kubernetes-e2e-trusty.yaml (hunk 2 above)
Comment (wonderfly) [inline hack/jenkins/job-configs/kubernetes-e2e-trusty.yaml]: Done

File: hack/jenkins/job-configs/kubernetes-e2e-trusty.yaml (hunk 1 above)
Comment (wonderfly) [inline hack/jenkins/job-configs/kubernetes-e2e-trusty.yaml]: Done

**Comment by wonderfly** [comment]
> I'd be happy to do this for you for these jobs.

That'll be great if you can help. :smile:

**Comment by spxtr** [comment]
LGTM. I've backed up the configs.

I can merge this manually tomorrow morning if that's fine with you.

**Comment by spxtr** [comment]
Ref #18122

**Comment by fejta** [comment]
FYI this is for https://github.com/kubernetes/kubernetes/pull/20620

**Comment by wonderfly** [comment]
> I can merge this manually tomorrow morning if that's fine with you.

SGTM. Thanks.

**Comment by wonderfly** [comment]
@spxtr I guess this is good to go after #20843 is merged?

**Comment by spxtr** [comment]
Manual merge to watch Jenkins.

**Pull request merged**